funcbench/Dockerfile

```diff
@@ -7,10 +7,18 @@ COPY go.* ./
 
 RUN CGO_ENABLED=0 GO111MODULE=on go build -a -tags netgo -o /go/bin/funcbench
 
-FROM golang:1.13-alpine
+FROM prominfra/prombench:master
 LABEL maintainer="The Prometheus Authors <prometheus-developers@googlegroups.com>"
 
-RUN apk add git
+WORKDIR /funcbench
+
+ENV PROMBENCH_CMD prombench
+
 COPY --from=builder /go/bin/funcbench /bin/
+COPY funcbench/ ./
+
+# Need 'eval' to prevent bash keywords be run as commands
+RUN echo -e '#!/bin/sh\ncd /funcbench\neval "$@"' >/usr/bin/docker_entrypoint
+RUN chmod u+x /usr/bin/docker_entrypoint
```

Comment:
Is `docker_entrypoint` necessary?

Reply:
I could not find a better way to do thing like this: https://github.com/prometheus/prometheus/blob/9e265aba10056547fd920fe603b31eecfced5966/.github/workflows/prombench.yml#L35

There possibly is a better way but for now that's the way we are passing these to prombench docker image(using `eval`), so yes we need some kind of script. we could rename it something else ofcourse.

Reply:
Is this method not good enough?
https://github.com/prometheus/test-infra/blob/9134f1ff2f5b9ebfec88e8e43c737fb71216bde7/pkg/provider/gke/gke.go#L386-L389

Reply:
OK, I noticed #119 , you mentioned a `Cluster in 'FailedPrecondition' state` error. But seems `until make ...` is not a good way to fix, what if the previous operation removed the nodepools? Then fall into a dead loop?

Reply:
> what if the previous operation removed the nodepools? Then fall into a dead loop?

`until make all_nodepools_running; do echo "waiting for nodepools to be created"; sleep 10; done;`
This basically says, only start deleting nodepools once all nodepools are running, otherwise the github actions that's creating nodepools might create another nodepool as they are running independently.

what do you mean by previous operation? two `/prombench cancel` sent at the same time? In that case one of the nodepool will start deleting first, once it starts doing that `all_nodepools_running` will be returning false so the second one will just keep looping unless some other command independently creates the nodepools. This is the same as running `prombench delete` on a nodepool that does not exist. I think the current waiting mechanism is okay.

We could have some kind of timeout for this but since we're running on Github actions it takes care of it (the 6hrs timeout) and it's very rare that that'll happen so we can probably not worry about it now.

The prow setup followed something like setting a label to the github PR and if that label is set, don't  start the test but the issue itself didn't mean that the benchmark truly is running or not, so we changed it to this.

If we want a more rigid check we can for some metric (maybe the namespace metric) but this approach is working fine for now.

> Cluster in `'FailedPrecondition' state`

That's a gke error, we can probably improve on the logging in the `pkg/gke` but it's not creating any problem, for us it just means we're in progress for the deletion process. The comment I made had to do something with the order how the resource and nodepools are deleted `make clean` the current order is fine and working so far with GKE.

Reply:
Thanks for the explanation @geekodour . I misunderstood the target `make all_nodepools_deleted`, which is not removing but checking. :P
Adding a label to the issue/PR sounds a good idea, we can track all the failed benchmarks before users notify us.

Back to the PR, I think we don't use `docker_entrypoint` here in funcbench, can we remove it here?

Reply:
Yes, we can remove them. I added them here for consistency between the docker images. I'll try simplifying this. thanks for the review :)